optimum/bettertransformer/models/encoder_models.py

```diff
@@ -751,3 +751,111 @@ def forward(self, hidden_states, attention_mask, **__):
         if hidden_states.is_nested and self.is_last_layer:
             hidden_states = hidden_states.to_padded_tensor(0.0)
         return (hidden_states,)
+
+
+class MobileBertLayerBetterTransformer(BetterTransformerBaseLayer):
+    def __init__(self, mobilebert_layer, config):
+        r"""
+        A simple conversion of the MobileBERT layer to its `BetterTransformer` implementation.
+
+        Args:
+            MobileBert_layer (`torch.nn.Module`):
+                The original MobileBERT Layer where the weights needs to be retrieved.
+        """
+        super().__init__(config)
+        # In_proj layer
+        self.in_proj_weight = nn.Parameter(
+            torch.cat(
+                [
+                    mobilebert_layer.attention.self.query.weight,
+                    mobilebert_layer.attention.self.key.weight,
+                    mobilebert_layer.attention.self.value.weight,
+                ]
+            )
+        )
+        self.in_proj_bias = nn.Parameter(
+            torch.cat(
+                [
+                    mobilebert_layer.attention.self.query.bias,
+                    mobilebert_layer.attention.self.key.bias,
+                    mobilebert_layer.attention.self.value.bias,
+                ]
+            )
+        )
+
+        # Out proj layer
+        self.out_proj_weight = mobilebert_layer.attention.output.dense.weight
+        self.out_proj_bias = mobilebert_layer.attention.output.dense.bias
+
+        # Linear layer 1
+        self.linear1_weight = mobilebert_layer.intermediate.dense.weight
+        self.linear1_bias = mobilebert_layer.intermediate.dense.bias
+
+        # Linear layer 2
+        self.linear2_weight = mobilebert_layer.output.dense.weight
+        self.linear2_bias = mobilebert_layer.output.dense.bias
+
+        # Layer norm 1
+        self.norm1_eps = mobilebert_layer.attention.output.LayerNorm.eps
+        self.norm1_weight = mobilebert_layer.attention.output.LayerNorm.weight
+        self.norm1_bias = mobilebert_layer.attention.output.LayerNorm.bias
+
+        # Layer norm 2
+        self.norm2_eps = mobilebert_layer.output.LayerNorm.eps
+        self.norm2_weight = mobilebert_layer.output.LayerNorm.weight
+        self.norm2_bias = mobilebert_layer.output.LayerNorm.bias
+
+        # Model hyper parameters
+        self.num_heads = mobilebert_layer.attention.self.num_attention_heads
+        self.embed_dim = mobilebert_layer.attention.self.all_head_size
+
+        # Last step: set the last layer to `False` -> this will be set to `True` when converting the model
+        self.is_last_layer = False
+
+        self.validate_bettertransformer()
+
+    def forward(self, hidden_states, attention_mask, *_):
```

Comment:
What is the purpose of `*_` in the end of the arguments?

Reply:
I was not sure, I saw *_ in every other forward call definition.

Reply:
Alright, maybe we should also add `**__` or something to handle potential keyword arguments?
@younesbelkada

Reply:
I saw both some forward def having both *_ and **_ . We should possibly add more information about it in the guide published.

Reply:
Yes basically it is to "handle" potential arguments that were passed to the original forward function, by allowing this forward function to get them, but ignoring them right after.